pkgs/hdf5/hdf5.yaml

```diff
@@ -18,15 +18,16 @@ build_stages:
   bash: |
     export CC=${MPICC}
 
-- when: machine == 'CrayXE6'
+- when: machine == 'CrayXC30'
```

Comment:
The hashdist's YAML domain specific  langauge understands basic logic, so if you make this `when: machine == 'CrayXE6' or machine == 'CrayXC30'`. Your changes will be backward compatible with the existing XE6 support.